ui/src/tool-panel/km-post/dialog/km-post-edit-store.ts

```diff
@@ -52,6 +53,61 @@ export const initialKmPostEditState: KmPostEditState = {
     allFieldsCommitted: false,
 };
 
+export const WGS_84_PROJECTION = '+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs';
+
+const SOUTHERNMOST_POINT_OF_FINLAND_DEG = 59.505;
+const NORTHMOST_POINT_OF_FINLAND_DEG = 70.092;
+const NORTHING_MARGIN_DEG = 0.1;
+const EASTING_MARGIN_BETWEEN_GKS_DEG = 0.01;
+
+// GK-FIN coordinate systems currently only used for the live display of layout coordinates when editing km post
+// positions manually
+export const GK_FIN_COORDINATE_SYSTEMS: [Srid, string][] = [...Array(12)].map(
+    (_, meridianIndex) => {
+        const meridian = 19 + meridianIndex;
+        const falseNorthing = meridian * 1e6 + 0.5e6;
+        const srid = `EPSG:${3873 + meridianIndex}`;
+        const projection = `+proj=tmerc +lat_0=0 +lon_0=${meridian} +k=1 +x_0=${falseNorthing} +y_0=0 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs`;
+        return [srid, projection];
+    },
+);
+
+export const isWithinEastingMargin = (point: GeometryPoint): boolean => {
+    const wgs84Point = pointToWgs84(point);
+    const gkIndex = GK_FIN_COORDINATE_SYSTEMS.findIndex((gk) => first(gk) === point.srid);
+    return (
+        !!wgs84Point &&
+        Math.abs(wgs84Point?.x - (gkIndex + 19)) < 0.5 + EASTING_MARGIN_BETWEEN_GKS_DEG
+    );
+};
+
+const pointToWgs84 = (point: GeometryPoint | undefined): Point | undefined => {
+    const currentPointProjection = point
+        ? GK_FIN_COORDINATE_SYSTEMS.find(([srid]) => srid === point.srid)?.[1]
+        : undefined;
+    return point && currentPointProjection
+        ? proj4(currentPointProjection, WGS_84_PROJECTION).forward({
+              x: point.x,
+              y: point.y,
+          })
+        : undefined;
+};
```

Comment:
Kaikki validoinnit tehdään WGS84:ssa, sillä se käyttää yksikköinään asteita, ja GK-kaistat on määritelty edellisen asteen puolivälistä seuraavaan (esim. GK24 kattaa alueen 23° 30'..24° 30'.)